modules/wallet/transactionbuilder.go

```diff
@@ -104,115 +104,77 @@ func calculateAmountFromOutputs(outputs []types.SiacoinOutput, fee types.Currenc
 	return amount
 }
 
-// FundSiacoinsForOutputs will add enough inputs to cover the outputs to be
-// sent in the transaction. In contrast to FundSiacoins, FundSiacoinsForOutputs
-// does not aggregate inputs into one output equaling 'amount' - with a refund,
-// potentially - for later use by an output or other transaction fee. Rather,
-// it aggregates enough inputs to cover the outputs, adds the inputs and outputs
-// to the transaction, and also generates a refund output if necessary. A miner
-// fee of 0 or greater is also taken into account in the input aggregation and
-// added to the transaction if necessary.
-func (tb *transactionBuilder) FundSiacoinsForOutputs(outputs []types.SiacoinOutput, fee types.Currency) error {
-	// dustThreshold has to be obtained separate from the lock
+func (tb *transactionBuilder) fund(amountToFund types.Currency, refundID types.SiacoinOutputID, inOutputs sortedOutputs) (types.Currency, []types.SiacoinOutputID, error) {
```

Comment:
You should comment this, since it's such a long function. It's also not immediately obvious to me what "inOutputs" means. Do you mean "spendableOutputs"?